File: app/src/main/java/com/djfos/im/filter/FilterThreshold.java
package com.djfos.im.filter;

import org.opencv.core.Mat;
import org.opencv.imgproc.Imgproc;

public class FilterThreshold extends BaseFilter{
    public int threshold;
    FilterThreshold(){
        type = FilterType.THRESHOLD;
    }
    @Override
    Mat apply(Mat in) {
        Mat mat = new Mat();
        Imgproc.threshold(in, mat, threshold, 255, Imgproc.THRESH_BINARY);
        return mat;
    }
}

File: app/src/main/java/com/djfos/im/filter/FilterThreshold.kt
package com.djfos.im.filter

import org.opencv.core.Mat
import org.opencv.imgproc.Imgproc


class FilterThreshold : AbstractFilter() {
    override val type = FilterType.Thresh

    var threshold: Int = 125

    override fun apply(input: Mat): Mat {
        val mat = Mat()
        Imgproc.threshold(input, mat, threshold.toDouble(), 255.0, Imgproc.THRESH_BINARY)
        return mat
    }
}
